packages/marketplace/src/sagas/settings/settings.ts

```diff
@@ -0,0 +1,46 @@
+import { put, fork, all, call, takeLatest } from '@redux-saga/core/effects'
+
+import { Action } from '@/types/core'
+import ActionTypes from '@/constants/action-types'
+import { ChangePasswordParams, settingLoadingVisibility } from '@/actions/settings'
+import { reapitConnectBrowserSession } from '@/core/connect-session'
+import { changePasswordService } from '@/services/cognito-identity'
+import { notification } from '@reapit/elements'
+
+export const clientPasswordChange = function*({ data }: Action<ChangePasswordParams>) {
+  yield put(settingLoadingVisibility(true))
+  try {
+    /* rename for compatible reason */
+    const { currentPassword: password, password: newPassword, email } = data
+    const cognitoClientId = window.reapit.config.cognitoClientId
+    const response = yield call(changePasswordService, {
+      userName: email,
+      password,
+      newPassword,
+      cognitoClientId,
+    })
+    const isCallAPISuccess = response === 'SUCCESS'
+    if (!isCallAPISuccess) {
+      throw new Error('Server error')
+    }
+    localStorage.setItem('isPasswordChanged', 'true')
```

Comment:
Sorry, bit confused why this is necessary - why are we saving to localStorage here?

Reply:
@willmcvay use this value as a flag to show the notification on login screen.
Use it here https://github.com/reapit/foundations/blob/feat/2333-add-change-password-to-setting-page-on-marketplace/packages/marketplace/src/components/pages/login/login.tsx#L29

Reply:
Cool, do we clear local storage afterwards? Or the user will see the notification forever!

Reply:
@willmcvay after display notification message, we remove the local storage value.
https://github.com/reapit/foundations/blob/feat/2333-add-change-password-to-setting-page-on-marketplace/packages/marketplace/src/components/pages/login/login.tsx#L18

Reply:
Great, just checking!